LanguageFeatures/Augmentation-libraries/extensions_A02_t01.dart

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Extension declarations can be augmented:
+///
+/// extensionDeclaration ::=
+///     'extension' typeIdentifierNotType? typeParameters? 'on' type
+///     memberedDeclarationBody
+///   | 'augment' 'extension' typeIdentifierNotType typeParameters?
+///     memberedDeclarationBody
+///
+/// Note that only extensions with names allow a leading augment. Since
+/// augmentations are matched with their introductory declaration by name,
+/// unnamed extensions can't be augmented.
+///
+/// @description Checks that it is a compile-time error to augment an unnamed
+/// extension declaration.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=augmentations
+
+class C {
+}
+
+extension on C {
+  int zero() => 0;
+}
+
+augment extension on C {
```

Comment:
The name cannot be omitted here, so the name will be `on`, but there is no extension whose name is `on` that we can augment. Also, the `C` that follows the word `on` is a syntax error.

Perhaps just try to augment the nameless extension without the `on` stuff?:
```suggestion
augment extension {
```

Reply:
It's a syntax error anyway, so we basically just have to accept that `augment extension { ...` won't work, and the parser doesn't really care _why_ it doesn't work. ;-)

Reply:
Updated.